ENIGMAsystem/SHELL/Universal_System/Extensions/Json/short_json_converter.cpp

```diff
@@ -0,0 +1,548 @@
+        virtual_pointer_ += temp_pointer_ - pointer_ - 1;
+        pointer_ += temp_pointer_ - pointer_ - 1;
+        return true;
+      }
+      case '.': {
+        if (is_double_)
+          return false;
+        else
+          is_double_ = true;
+        break;
+      }
+      case '0':
+      case '1':
+      case '2':
+      case '3':
+      case '4':
+      case '5':
+      case '6':
+      case '7':
+      case '8':
+      case '9':
+        break;
+      default:
+        return false;
+    }
+    temp_pointer_++;
+  }
+
+  return false;
+}  // accumulate_number_value
+
+bool ShortJSONConverter::accumulate_boolean_value() {
+  size_t temp_pointer_{pointer_};
+
+  switch (data_.at(temp_pointer_)) {
+    case 't':
+      if (temp_pointer_ + 3 >= data_.length()) return false;
+
+      if (data_.at(temp_pointer_ + 1) != 'r' || data_.at(temp_pointer_ + 2) != 'u' ||
+          data_.at(temp_pointer_ + 3) != 'e')
+        return false;
+
+      temp_pointer_ += 4;  // "true" length is 4.
+
+      break;
+    case 'f':
+      if (temp_pointer_ + 4 >= data_.length()) return false;
+
+      if (data_.at(temp_pointer_ + 1) != 'a' || data_.at(temp_pointer_ + 2) != 'l' ||
+          data_.at(temp_pointer_ + 3) != 's' || data_.at(temp_pointer_ + 4) != 'e')
+        return false;
+
+      temp_pointer_ += 5;  // "false" length is 5.
+
+      break;
+    default:
+      return false;
+  }
+
+  if (temp_pointer_ >= data_.length()) return false;
+
+  accumulate_single_key();
+  levels_accumulators_.top() += data_.substr(pointer_, temp_pointer_ - pointer_);
+  virtual_pointer_ += temp_pointer_ - pointer_ - 1;
+  pointer_ += temp_pointer_ - pointer_ - 1;
+
+  return true;
+}  // accumulate_boolean_value
+
+bool ShortJSONConverter::accumulate_null_value() {
+  size_t temp_pointer_{pointer_};
+
+  if (temp_pointer_ + 3 >= data_.length()) return false;
+
+  if (data_.at(temp_pointer_ + 1) != 'u' || data_.at(temp_pointer_ + 2) != 'l' || data_.at(temp_pointer_ + 3) != 'l')
+    return false;
+
+  temp_pointer_ += 4;  // "null" length is 4.
+
+  if (temp_pointer_ >= data_.length()) return false;
+
+  accumulate_single_key();
+  levels_accumulators_.top() += data_.substr(pointer_, temp_pointer_ - pointer_);
+  virtual_pointer_ += temp_pointer_ - pointer_ - 1;
+  pointer_ += temp_pointer_ - pointer_ - 1;
+
+  return true;
+}  // accumulate_null_value
+
+void ShortJSONConverter::map_short_json_indices() {
+  size_t number_of_entities_{levels_entities_boundaries_pointers_.top().size()};
+  size_t shift_left_{levels_entities_boundaries_pointers_.top().front().first};
+
+  size_t offset_{0};
+
+  /*
+    1. Map the front boundaries.
+    2. Calculate the new offset, save it in "offset_", pop it from front.
+    3. Push the mapped boundaries to the back of the queue, pop it from front.
+    4. Repeat steps 1 - 3.
+  */
+  for (size_t i{0}; i < number_of_entities_; i++) {
+    levels_entities_boundaries_pointers_.top().front().first -= shift_left_;
```

Comment:
You should probably bind `levels_entities_boundaries_pointers_.top()` to a reference variable with a shorter name so that you don't have to repeat the whole expression every time, something like:
```cpp
    auto &top = levels_entities_boundaries_pointers_.top();
    top.front().first -= shift_left_;
    top.front().second -= shift_left_;

    top..front().first += offset_;
    top.front().second += offset_ + levels_entities_offsets_.top().front();

    offset_ += levels_entities_offsets_.top().front();

    levels_entities_offsets_.top().pop();

    top.push(levels_entities_boundaries_pointers_.top().front());
    top.pop();
```